crates/holochain_types/src/app.rs

```diff
@@ -475,25 +520,63 @@ impl InstalledAppCommon {
         &self.role_assignments
     }
 
-    /// Add a cloned cell
-    pub fn add_clone(&mut self, role_id: &AppRoleId, cell_id: CellId) -> AppResult<()> {
-        let role = self.role_mut(role_id)?;
+    /// Get the next clone index for a role id.
+    pub fn next_clone_index(&self, role_id: &AppRoleId) -> AppResult<u32> {
+        Ok(self.role(role_id)?.next_clone_index)
+    }
+
+    /// Increment index for next clone cell for a role id.
+    pub fn increment_next_clone_index(&mut self, role_id: &AppRoleId) -> AppResult<()> {
+        self.role_mut(role_id)?.next_clone_index += 1;
+        Ok(())
+    }
```

Comment:
Could be nice if these were rolled into a single pre-increment function that returns the current index after incrementing it, since we always do those two things together. But this is fine too.

Reply:
I agree and would have done this if it happened in the same place. I'm awaiting your comment to the possible issue of an orphan DNA in the DB.